docs/dev/libvirt/README.md

```diff
@@ -91,67 +91,89 @@ components drive deployment of worker machines.  The libvirt cluster-api
-tcp_port = "16509"
-```
+#### For systemd activated libvirt
+
+This applies only if the libvirt daemon is configured to use socket activation.
+This is currently the case on Fedora 31 (and later) and Arch Linux.
 
-On Fedora 31, you also need to enable and start the libvirtd TCP
-socket, which is managed by systemd:
+First, you need to start the libvirtd TCP socket, which is managed by systemd:
 
 ```sh
-sudo systemctl enable libvirtd-tcp.socket
 sudo systemctl start libvirtd-tcp.socket
 ```
 
-after which you need to restart libvirtd.
+To make this change persistent accross reboots you can optionally enable it:
 
-**NOTE:** that the above configuration disables all encryption and
-authentication options in libvirtd and causes it to listen on all
-network interfaces and IP addresses. **A connection to this privileged
-libvirtd gives the client privileges equivalent to those of a root
-shell.** This configuration has a security impact on a par with
-running a telnet server with no root password set. It is critical
-to follow the steps below to **configure the firewall to prevent
-access to libvirt from other hosts on the LAN/WAN**.
+```sh
+sudo systemctl enable libvirtd-tcp.socket
+```
 
+Then to enable TCP access to libvirtd, modify `/etc/libvirt/libvirtd.conf` and
+set the following:
 
-#### Configure qemu.conf
+```
+auth_tcp = "none"
+```
 
-On Debian/Ubuntu it might be needed to configure security driver for qemu.
-Installer uses terraform libvirt, and it has a known issue, that might cause
-unexpected `Could not open '/var/lib/libvirt/images/<FILE_NAME>': Permission denied`
-errors. Double check that `security_driver = "none"` line is present in 
-`/etc/libvirt/qemu.conf` and not commented out.
+Then restart libvirt:
 
-#### Configure the service runner to pass `--listen` to libvirtd
+```sh
+sudo systemctl restart libvirtd
+```
+
+#### For permanently running libvirt daemon
 
-**NOTE:** if the installation of libvirt included support for socket
-activation via the `libvirt-tcp.socket` systemd unit, the `--listen`
-argument should not be added and thus this step can be skipped.
+This applies only if the libvirt daemon is configured to start and listen on
+UNIX/TCP sockets by itself and not through socket activation.
 
-If socket activation is not available, you'll have to pass an additional
-command-line argument to libvirtd. On Red Hat based distros, modify
-`/etc/sysconfig/libvirtd` and set:
+For RHEL/CentOS, make sure that the following is set in
+`/etc/sysconfig/libvirtd`:
 
 ```
 LIBVIRTD_ARGS="--listen"
 ```
 
-On Debian based distros, modify `/etc/default/libvirtd` and set:
+For Debian based distros, make sure that the following is set in
+`/etc/default/libvirtd`:
 
 ```
 libvirtd_opts="--listen"
 ```
 
-Next, restart libvirt: `systemctl restart libvirtd`
+Then to enable TCP access to libvirtd, modify `/etc/libvirt/libvirtd.conf` and
+set the following:
+
+```
+listen_tls = 0
+listen_tcp = 1
+auth_tcp = "none"
+tcp_port = "16509"
+```
```

Comment:
I'm still a bit dubious with the splitting of the libvirtd.conf instructions, even if they are ignored, it's not harmful to set them in socket-activation setups

Reply:
It's not harmful but it's also unnecessary and might trigger questions. I think we need to have two clearly separated cases to avoid confusion and keeping each case to the point helps with that.